contracts/WildcardSteward_v3.sol

```diff
@@ -540,65 +554,32 @@ contract WildcardSteward_v3 is Initializable {
     }
 
     function _collectPatronage(uint256 tokenId) public {
-        if (state[tokenId] == StewardState.Owned) {
-            address tokenPatron = assetToken.ownerOf(tokenId);
-
-            uint256 patronageOwedByTokenPatron = patronageOwedPatron(
-                tokenPatron
-            );
-
-            uint256 timeSinceLastMint;
-
-            if (
-                patronageOwedByTokenPatron > 0 &&
-                patronageOwedByTokenPatron >= deposit[tokenPatron]
-            ) {
-
-                    uint256 previousCollectionTime
-                 = timeLastCollectedPatron[tokenPatron];
-                uint256 newTimeLastCollected = previousCollectionTime.add(
-                    (
-                        (now.sub(previousCollectionTime))
-                            .mul(deposit[tokenPatron])
-                            .div(patronageOwedByTokenPatron)
-                    )
-                );
-                timeLastCollectedPatron[tokenPatron] = newTimeLastCollected;
-                deposit[tokenPatron] = 0;
-                timeSinceLastMint = (
-                    newTimeLastCollected.sub(previousCollectionTime)
-                );
+        address tokenPatron = assetToken.ownerOf(tokenId);
+        uint256 newTimeLastCollectedOnForeclosure = _collectPatronagePatron(
+            tokenPatron
+        );
 
-                // The bellow 3 lines are the main difference between this function and the `_collectPatronagePatron` function.
-                _collectLoyaltyPatron(tokenPatron, timeSinceLastMint); // NOTE: you have to call collectLoyaltyPatron before collecting your deposit.
-                tokenAuctionBeginTimestamp[tokenId] = newTimeLastCollected + 1; // The auction starts the second after the last time collected.
-                _foreclose(tokenId);
-
-                address benefactor = benefactors[tokenId];
-                uint256 amountOverCredited = price[tokenId]
-                    .mul(
-                    (
-                        timeLastCollectedBenefactor[benefactor].sub(
-                            newTimeLastCollected
-                        )
-                    )
+        bool tokenForeclosed = newTimeLastCollectedOnForeclosure > 0;
```

Comment:
You can just use timeLastCollectedPatron[tokenPatron], 
which is set in the other function, I don't think you need to return this value from the other function

Reply:
Yeah, I mentioned this on slack. Not sure if the map look-up is more expensive gas wise